libraries/types/pbft_block/src/period_data.cpp

```diff
@@ -26,15 +26,17 @@ PeriodData::PeriodData(dev::RLP&& rlp) {
     dag_blocks.emplace_back(dag_block_rlp);
   }
 
-  for (auto const trx_rlp : *it) {
+  for (auto const trx_rlp : *it++) {
     transactions.emplace_back(std::make_shared<Transaction>(trx_rlp));
   }
+
+  bonus_votes_count = (*it).toInt<size_t>();
```

Comment:
this info is inside previous_block_cert_votes, isnt it ?

Reply:
yes, but I need to save it to DB or re-validate all votes, not sure what is better

Reply:
this is related to the previous comment right ? You would need to revalidate few blocks only after "node restart" ? If so, I think revalidating is better

Reply:
yes, this will be removed